src/pybind/mgr/dashboard/controllers/rbd_mirroring.py

```diff
@@ -45,146 +45,160 @@ def composed_decorator(func):
+                'version': metadata['ceph_version'],
+                'server_hostname': hostname,
+                'service': service,
+                'server': server,
+                'metadata': metadata,
+                'status': status
+            }
+            daemon = dict(daemon, **get_daemon_health(daemon))
+            daemons.append(daemon)
+
+    return sorted(daemons, key=lambda k: k['instance_id'])
+
+
+def get_daemon_health(daemon):
+    health = {
+        'health_color': 'info',
+        'health': 'Unknown'
+    }
+    for _, pool_data in daemon['status'].items():
+        if (health['health'] != 'error'
+                and [k for k, v in pool_data.get('callouts', {}).items()
+                     if v['level'] == 'error']):
+            health = {
+                'health_color': 'error',
+                'health': 'Error'
+            }
+        elif (health['health'] != 'error'
+                and [k for k, v in pool_data.get('callouts', {}).items()
+                     if v['level'] == 'warning']):
+            health = {
+                'health_color': 'warning',
+                'health': 'Warning'
+            }
+        elif health['health_color'] == 'info':
+            health = {
+                'health_color': 'success',
+                'health': 'OK'
+            }
+    return health
+
+
+def get_pools(daemons):  # pylint: disable=R0912, R0915
+    pool_names = [pool['pool_name'] for pool in CephService.get_pool_list('rbd')
+                  if pool.get('type', 1) == 1]
+    pool_stats = _get_pool_stats(pool_names)
+    _update_pool_stats(daemons, pool_stats)
+    return pool_stats
+
+
+def _update_pool_stats(daemons, pool_stats):
+    _update_pool_stats_with_daemons(daemons, pool_stats)
+    for _, stats in pool_stats.items():
+        if stats['mirror_mode'] == 'disabled':
+            continue
+        if stats.get('health', None) is None:
+            # daemon doesn't know about pool
+            stats['health_color'] = 'error'
+            stats['health'] = 'Error'
+        elif stats.get('leader_id', None) is None:
+            # no daemons are managing the pool as leader instance
+            stats['health_color'] = 'warning'
+            stats['health'] = 'Warning'
+
+
+def _update_pool_stats_with_daemons(daemons, pool_stats):
+    for daemon in daemons:
+        for _, pool_data in daemon['status'].items():
+            stats = pool_stats.get(pool_data['name'], None)  # type: ignore
+            if stats is None:
                 continue
-            if stats.get('health', None) is None:
-                # daemon doesn't know about pool
+
+            if pool_data.get('leader', False):
+                # leader instance stores image counts
+                stats['leader_id'] = daemon['metadata']['instance_id']
+                stats['image_local_count'] = pool_data.get('image_local_count', 0)
+                stats['image_remote_count'] = pool_data.get('image_remote_count', 0)
```

Comment:
I think this if is unreachable looking at get_pool_stats

Reply:
You mean the `if stats.get('health', None) is None:`?

Reply:
If I remember correctly it was because in `_get_pool_stats` (called in `get_pools` before `update_pool_stats_with_dameons`) it never adds a `leader` key to the `stats` dict.